### PR TITLE
sql: fix error when use true in check constraint expression

### LINKED ISSUE #44689: Unexpected error when `CHECK` a predicate that is always true.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

The following query triggers an error:
```
CREATE TABLE t0(c1 NUMERIC CHECK (true));
```

This is the error message:
```
Column check constraint 't0_chk_1' references other column.
```
A `true` predicate means no value will violate this constraint. This query can triggers this error in MySQL, and MySQL verified this is a bug.
https://bugs.mysql.com/bug.php?id=111421

### 2. What did you expect to see? (Required)
No error.
### 3. What did you see instead (Required)
```
Column check constraint 't0_chk_1' references other column.
```

### 4. What is your TiDB version? (Required)

```
Release Version: v7.2.0-alpha-421-gabd9ff5ca6\nEdition: Community\nGit Commit Hash: abd9ff5ca6fe866224f243c931844f050fb4952c\nGit Branch: master\nUTC Build Time: 2023-06-14 08:57:42\nGoVersion: go1.20.3\nRace Enabled: false\nCheck Table Before Drop: false\nStore: unistore
```

### PR DESCRIPTION
**Pull request by fzzf678**
### What problem does this PR solve?

Issue Number: close #44689 

Problem Summary:
As issue describe, when use `check(true)` in column check constraint, we receive unexpected error. The reason is when validate the column if exists, we don't consider this situation.

### What is changed and how it works?
Involve this situation.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by CbcWestwolf — APPROVED** [review]
LGTM

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -1997,13 +1997,21 @@ func BuildTableInfo(
...
-				if len(dependedColsMap) != 1 {
+				if len(dependedColsMap) > 1 {
 					return nil, dbterror.ErrColumnCheckConstraintReferOther.GenWithStackByArgs(constr.Name)
 				}
-				if _, ok := dependedColsMap[constr.InColumnName]; !ok {
-					return nil, dbterror.ErrColumnCheckConstraintReferOther.GenWithStackByArgs(constr.Name)
+				if len(dependedColsMap) == 0 {
+					// If dependedCols is empty, the expression must be true/false.
+					valExpr, ok := constr.Expr.(*driver.ValueExpr)
+					if !ok || !mysql.HasIsBooleanFlag(valExpr.GetType().GetFlag()) {
```
Comment (TonsnakeLin) [inline ddl/ddl_api.go:2006]: 可以添加个 check(false), check(1), check(-1) 用例

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -1997,13 +1997,21 @@ func BuildTableInfo(
 				}
 			} else {
 				// Check the column-type constraint dependency.
-				if len(dependedColsMap) != 1 {
+				if len(dependedColsMap) > 1 {
 					return nil, dbterror.ErrColumnCheckConstraintReferOther.GenWithStackByArgs(constr.Name)
 				}
-				if _, ok := dependedColsMap[constr.InColumnName]; !ok {
-					return nil, dbterror.ErrColumnCheckConstraintReferOther.GenWithStackByArgs(constr.Name)
+				if len(dependedColsMap) == 0 {
```
[the remark this answers is not in the document]
Comment (fzzf678) [reply ddl/ddl_api.go:2003]: emm, how?

[the remark this answers is not in the document]
Comment (CbcWestwolf) [reply ddl/ddl_api.go:2003]:
```go
				// Check the column-type constraint dependency.
				if len(dependedColsMap) > 1 {
					return nil, dbterror.ErrColumnCheckConstraintReferOther.GenWithStackByArgs(constr.Name)
				} else if len(dependedColsMap) == 0 {
					// If dependedCols is empty, the expression must be true/false.
					valExpr, ok := constr.Expr.(*driver.ValueExpr)
					if !ok || !mysql.HasIsBooleanFlag(valExpr.GetType().GetFlag()) {
						return nil, errors.Trace(errors.New("unsupported expression in check constraint"))
					}
				} else {
					if _, ok := dependedColsMap[constr.InColumnName]; !ok {
						return nil, dbterror.ErrColumnCheckConstraintReferOther.GenWithStackByArgs(constr.Name)
					}
					dependedCols = []model.CIStr{model.NewCIStr(constr.InColumnName)}
				}
```

**Pull request merged**